### PR TITLE
test: Run HTTP Server TCK for Azure HTTP Function

### PR DESCRIPTION
**Pull request by timyates**
Supersedes  #412 on the 4.0.x branch

### CODE REVIEW

**Comment by timyates** [comment]
Ok...so this is failing... but so is the master branch...

I'm off to fix that first...

**Comment by timyates** [comment]
@sdelamo disabled failing tests, and raised https://github.com/micronaut-projects/micronaut-azure/issues/439

File: test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/HttpRequestMessageAdapter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.microsoft.azure.functions.HttpStatus;
+import com.microsoft.azure.functions.HttpStatusType;
+
+import java.net.URI;
+import java.util.Map;
+
+public class HttpRequestMessageAdapter implements HttpRequestMessage {
+    @Override
+    public URI getUri() {
+        return null;
```
Comment (sdelamo) [inline test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/HttpRequestMessageAdapter.java:15]: is this normal

File: test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/AzureFunctionHttpServerUnderTest.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            result.append(URLEncoder.encode(params.get(k).toString(), "UTF-8"));
+        }
+        return result.toString();
+    }
+
+
+    private final static List<String> HEADERS_USED_IN_TEST_SUITE = Arrays.asList("X-Test-Filter",
+        "X-Captured-Remote-Address");
+
+    private final static List<String> STANDARD_HEADERS = Arrays.asList(
```
Comment (sdelamo) [inline test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/AzureFunctionHttpServerUnderTest.java:140]: This should be in core?

**Review by sdelamo — APPROVED** [review]

File: test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/HttpRequestMessageAdapter.java (hunk 1 above)
Comment (timyates) [reply test-suite-http-server-tck-azure-function-http/src/test/java/io/micronaut/http/server/tck/azurehttpfunction/HttpRequestMessageAdapter.java:15]: I'm not sure...  You added that on 19th December... Was I supposed to alter it?

**Pull request merged**